ads/atomx.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, validateDataExists} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function atomx(global, data) {
+  const args = [];
+  args.push('size=' + data.width + 'x' + data.height);
+
+  // Validate empty fields
+  validateDataExists(data, ['id']);
+
+  args.push('id=' + encodeURIComponent(data.id));
+
+  if (data.click) {
+    args.push('click=' + encodeURIComponent(data.click));
+  }
+  if (data.uv1) {
+    args.push('uv1=' + encodeURIComponent(data.uv1));
+  }
+  if (data.uv2) {
+    args.push('uv2=' + encodeURIComponent(data.uv2));
+  }
+  if (data.uv3) {
+    args.push('uv3=' + encodeURIComponent(data.uv3));
+  }
+  if (data.context) {
+    args.push('context=' + encodeURIComponent(data.context));
+  }
```

Comment:
I'm not strong about this, but you can probably make them in a loop since they all have the same pattern.